docs/src/reference/variables.md

```diff
@@ -2,6 +2,18 @@
 
 Variables are part of a workflow instance and represent the data of the instance. A variable has a name and a JSON value. The visibility of a variable is defined by its variable scope.
 
+## Variable Names
+
+The name of a variable can be any alphanumeric string including the `_` symbol. For a combination of words, it is recommended to use the `camelCase` or the `snake_case` format.
+
+When accessing a variable in an expression, keep in mind that the variable name is case-sensitive.
+
+Restrictions of a variable name:
+* it must not start with a number
+* it must not contain whitespaces
+* it must not contain an operator (e.g. `+`, `-`, `*`, `/`, `=`, `>`, `?`, `.`)
+* it must not be a literal (`null`, `true`, `false`) or a keyword (e.g. `function`, `if`, `then`, `else`, `for`, `between`, `instance`, `of`, `not`)
```

Comment:
I'm not entirely sure, but I'm wondering whether `it may not contain x` is better for these restrictions.